ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/CliSandboxOnXRunner.scala

```diff
@@ -5,80 +5,64 @@ package com.daml.ledger.sandbox
-        c.maxDeduplicationDuration.forall(_.compareTo(Duration.ofHours(1L)) <= 0),
-        (),
-        "Maximum supported deduplication duration is one hour",
-      )
-    )
-    ()
+  def program[T](acquireResource: ResourceContext => Resource[T]): Unit = {
+    new ProgramResource(ResourceOwner.forResource(acquireResource))
+      .run(ResourceContext.apply)
   }
 
-  def owner(
+  def run(
       args: collection.Seq[String],
       manipulateConfig: CliConfig[BridgeConfig] => CliConfig[BridgeConfig] = identity,
-  ): ResourceOwner[Unit] =
+  ): Unit =
     CliConfig
-      .owner(
-        RunnerName,
-        bridgeConfigParser,
-        BridgeConfig.Default,
-        args,
-      )
+      .parse(RunnerName, BridgeConfig.Parser, BridgeConfig.Default, args)
       .map(manipulateConfig)
-      .flatMap(owner)
+      .foreach(runProgram)
 
-  def owner(originalConfig: CliConfig[BridgeConfig]): ResourceOwner[Unit] =
-    new ResourceOwner[Unit] {
-      override def acquire()(implicit context: ResourceContext): Resource[Unit] = {
-        Banner.show(Console.out)
+  private def runProgram(config: CliConfig[BridgeConfig]): Unit =
+    config.mode match {
+      case Mode.Run =>
+        SandboxOnXConfig
+          .loadFromConfig(config.configFiles, config.configMap)
+          .fold(
+            System.err.println,
+            { sandboxOnXConfig =>
+              program(implicit context => runSandboxOnX(new BridgeConfigAdaptor, sandboxOnXConfig))
+            },
+          )
+      case Mode.DumpIndexMetadata(jdbcUrls) =>
+        program(implicit context => DumpIndexMetadata(jdbcUrls))
+      case Mode.ConvertConfig =>
+        Console.out.println(
+          ConfigRenderer.render(SandboxOnXConfig(new BridgeConfigAdaptor, config))
+        )
+      case Mode.RunLegacy =>
         val configAdaptor: BridgeConfigAdaptor = new BridgeConfigAdaptor
-        originalConfig.mode match {
-          case Mode.DumpIndexMetadata(jdbcUrls) =>
-            DumpIndexMetadata(jdbcUrls)
-            sys.exit(0)
-          case Mode.Run =>
-            val config = CliConfigConverter.toConfig(configAdaptor, originalConfig)
-            run(
-              configAdaptor,
-              config,
-              originalConfig.extra.copy(maxDeduplicationDuration =
-                originalConfig.maxDeduplicationDuration.getOrElse(
-                  BridgeConfig.DefaultMaximumDeduplicationDuration
-                )
-              ),
-            ).map(_ => ())
-        }
-      }
+        val sandboxOnXConfig: SandboxOnXConfig = SandboxOnXConfig(configAdaptor, config)
+        program(implicit context => runSandboxOnX(configAdaptor, sandboxOnXConfig))
     }
+
+  private def runSandboxOnX(
+      configAdaptor: BridgeConfigAdaptor,
+      sandboxOnXConfig: SandboxOnXConfig,
+  )(implicit context: ResourceContext): Resource[Unit] = {
+    Banner.show(Console.out)
+    logger.withoutContext.info(s"Config used: ${ConfigRenderer.render(sandboxOnXConfig)}")
```

Comment:
Do we want to print the config used always at start-up? While I see it can be useful, it might be worth to have it on DEBUG level